src/platform/silabs/multi-ota/SiWx917/OTAWiFiFirmwareProcessor.cpp

```diff
@@ -0,0 +1,120 @@
+/*
+ *
+ *    Copyright (c) 2025 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include <platform/internal/CHIPDeviceLayerInternal.h>
+#include <platform/silabs/multi-ota/OTAMultiImageProcessorImpl.h>
+#include <platform/silabs/multi-ota/SiWx917/OTAWiFiFirmwareProcessor.h>
+
+#include <app/clusters/ota-requestor/OTARequestorInterface.h>
+#include <platform/silabs/SilabsConfig.h>
+#ifdef __cplusplus
+extern "C" {
+#endif
+#include "sl_si91x_driver.h"
+#ifdef __cplusplus
+}
+#endif
+
+namespace chip {
+
+CHIP_ERROR OTAWiFiFirmwareProcessor::ProcessInternal(ByteSpan & block)
+{
```

Comment:
We should try to avoid memcpy function and use the mutablebytespan instead since the input is a bytespan. This would somewhat simplify the logic and validation.

Reply:
I replaced the raw buffer handling and memcpy with bytespan operations.

Reply:
Mathieu, I understand that I have resolved your your comment by replacing the `memcpy `with `bytespan`. Could you please clarify exactly how this is to be changed?

Reply:
The goal of the comment was to use the Span.h APIs including the memcpy function as whole. To address the comment, you need to remove memcpy call as well.

Reply:
I have removed memcpy and used subSpan() API's.